dao/src/main/java/org/thingsboard/server/dao/ThingsboardPostgreSQLDialect.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright © 2016-2023 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao;
+
+import org.hibernate.dialect.function.SQLFunctionTemplate;
+import org.hibernate.type.BooleanType;
+
+public class ThingsboardPostgreSQLDialect extends org.hibernate.dialect.PostgreSQL10Dialect {
+    public ThingsboardPostgreSQLDialect() {
+        super();
+        registerFunction("ilike", new SQLFunctionTemplate(BooleanType.INSTANCE, "(?1 ILIKE ?2)"));
```

Comment:
Should `ilike` always be in lowercase or can we use `ILIKE` too?

Reply:
we can use ILIKE too

Reply:
Please provide an example of the SQL statement that is generated based on @Query and this dialect. I would like to double check that the final SQL statement uses ILIKE without any overhead

Reply:
SQL test property

```
spring.jpa.show-sql=true
spring.jpa.properties.hibernate.format_sql=true
```

I believe 
```
registerFunction("ilike", new SQLFunctionTemplate(StandardBasicTypes.BOOLEAN, "?1 ILIKE ?2"));
```

It is an excessive parenthesis.
```
Hibernate: 
    select
        deviceenti0_.id as id1_18_,
        deviceenti0_.created_time as created_2_18_,
        deviceenti0_.additional_info as addition3_18_,
        deviceenti0_.customer_id as customer4_18_,
        deviceenti0_.device_data as device_d5_18_,
        deviceenti0_.device_profile_id as device_p6_18_,
        deviceenti0_.external_id as external7_18_,
        deviceenti0_.firmware_id as firmware8_18_,
        deviceenti0_.label as label9_18_,
        deviceenti0_.name as name10_18_,
        deviceenti0_.software_id as softwar11_18_,
        deviceenti0_.tenant_id as tenant_12_18_,
        deviceenti0_.type as type13_18_ 
    from
        device deviceenti0_ 
    where
        deviceenti0_.tenant_id=? 
        and (
            (
                deviceenti0_.name ILIKE ('%'||?||'%')
            )=true 
            or (
                deviceenti0_.label ILIKE ('%'||?||'%')
            )=true
        ) 
    order by
        deviceenti0_.id asc nulls last limit ?
```

Reply:
without parenthesis
```
    from
        device deviceenti0_ 
    where
        deviceenti0_.tenant_id=? 
        and (
            deviceenti0_.name ILIKE ('%'||?||'%')=true 
            or deviceenti0_.label ILIKE ('%'||?||'%')=true
        ) 
    order by
        deviceenti0_.id asc nulls last limit ?
```